Library/Homebrew/formula.rb

```diff
@@ -553,13 +562,33 @@ def print_tap_action options={}
 
   # True if this formula is provided by Homebrew itself
   def core_formula?
-    path == Formula.path(name)
+    path == Formula.core_path(name)
   end
 
-  def self.path name
+  def self.core_path name
     Pathname.new("#{HOMEBREW_LIBRARY}/Formula/#{name.downcase}.rb")
   end
 
+  def self.path name
+    p = Formula.core_path(name)
+    return p if p.file?
+    name = name.downcase
+    taps = Dir["#{HOMEBREW_LIBRARY}/Taps/*/*/"]
+    p = taps.map { |tap| Dir["#{tap}#{name}.rb", "#{tap}Formula/#{name}.rb", "#{tap}HomebrewFormula/#{name}.rb"] }
+    p = p.flatten.map(&Pathname.method(:new)).select { |f| f.file? }
+    case p.size
+    when 0; nil
+    when 1; p[0]
+    else
+      ofail <<-EOS.undent
+        Multiple formulae found:
+        * #{p.map(&:to_s).join("\n        * ")}
+
+        Please use the fully-qualified name to refer the formula.
+      EOS
+    end
+  end
+
```

Comment:
Because, this method is mainly called by `Formulary.loader_for`. If the formula is not found here, it will try to find from Alias and cache.